src/Analyzers/CSharp/Analyzers/UseTupleSwap/CSharpUseTupleSwapDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseTupleSwap
+{
+    /// <summary>
+    /// Looks for code of the form:
+    /// 
+    /// <code>
+    ///     var temp = expr_a;
+    ///     expr_a = expr_b;
+    ///     expr_b = temp;
+    /// </code>
+    ///
+    /// and converts it to:
+    /// 
+    /// <code>
+    ///     (expr_b, expr_a) = (expr_a, expr_b);
+    /// </code>
+    /// 
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpUseTupleSwapDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public CSharpUseTupleSwapDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.UseTupleSwapDiagnosticId,
+                   EnforceOnBuildValues.UseTupleSwap,
+                   CSharpCodeStyleOptions.PreferTupleSwap,
```

Comment:
I asked before about code-style options that are semantically equivalent to just disabling the rule, and @sharwell [reply](https://github.com/dotnet/docs/issues/22450#issuecomment-765558476) was:

> The option sets the preferred style for other features that generate code automatically, even if the analyzer is disabled.

I don't believe this would be ever needed by any code-generation features. So why it exists? It's more distracting to the user IMO.

Reply:
it's a codestyleoption so we can set an appropriate editorconfig string for this.

Reply:
@CyrusNajmabadi Why do we need an editorconfig option if it's semantically identical to the compiler-supported `dotnet_diagnostic.IDE0180.severity = none`?

Reply:
Because the compielr supported form is terrible.  `dotnet_diagnostic.IDE0180.severity` means nothing to the user.  Whereas someone can actually go look at something like `csharp_prefer_tuple_swap` and understand what this setting corresponds to.  Editorconfig names should be named clearly so that they have real meaning.  Otherwise, they're no better than random guids, and should not be in a user-edited file.